### PR TITLE
enh(install): harden gorgone config generation

### PR DESCRIPTION
**Pull request by sc979**
## Description

Remove the gorgone configuration generation from the 20.04.0-beta.3 script
Add the generation in the 20.04.1 script
Use global variables and platform's constant instead of potentially modified files
Remove a configuration template no more used
Change the gorgone configuration template macros
Take in account these modifications in the install processes
Manage the configuration file duplication if already generated

**Fixes** # (MON-5391)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

Install or update a platform and check that the gorgone service start properly
Check the gorgone configuration
On a platform already updated to the 20.04.0, check that two new configuration files are generated :
/etc/centreon/config.d/10-database.yaml.new
/etc/centreon-gorgone/config.d/40-gorgoned.yaml.new

## Checklist

- [x] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/install/php/Update-20.04.1.php [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+    // get engine command
+    $res = $pearDB->query(
+        "SELECT command_file FROM cfg_nagios cn
+        JOIN nagios_server ns ON ns.id = cn.nagios_id
+        WHERE localhost = '1'"
+    );
+    $engineCommand = $res->fetch()['command_file'];
+
+    // set default values used by gorgone
```
Comment (sc979) [inline www/install/php/Update-20.04.1.php:48]: Wrong comment

File: www/install/php/Update-20.04.1.php [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        password,
+        db,
+        dbcstg,
+        _CENTREON_VARLIB_,
+        '/var/spool/centreon',
+        '/etc/snmp/centreon_traps',
+        '0.0.0.0',
+        '8085',
+        'false',
+        '/var/lib/centreon-gorgone',
```
Comment (sc979) [inline www/install/php/Update-20.04.1.php:80]: Line 80 et 81 are revert

File: www/install/php/Update-20.04.1.php [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+     * if already exists, generate a new file
+     *
+     * @param string - path to the file
+     *
+     * @return string - corrected filename
+     */
+    function returnFinalFileName(string $destinationFile)
+    {
+        if (file_exists($destinationFile)) {
+            $destinationFile .= 'new';
```
Comment (sc979) [inline www/install/php/Update-20.04.1.php:95]: New -> .new

File: www/install/php/Update-20.04.1.php [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    if (!file_exists($fileTpl) || 0 === filesize($fileTpl)) {
+        $errorMessage = 'Gorgone configuration template is empty or missing';
+        throw new \InvalidArgumentException($errorMessage);
+    }
+    $content = file_get_contents($fileTpl);
+    $content = preg_replace($pattern, $userValues, $content);
+    $destinationFolder = _CENTREON_ETC_ . '/../centreon-gorgone';
+    $destinationFile = returnFinalFileName($destinationFolder . '/config.d/40-gorgoned.yaml');
+
+    // checking if mandatory /etc/centreon-gorgone and sub-folders already exists
```
Comment (sc979) [inline www/install/php/Update-20.04.1.php:132]: Subfolders -> subfolder

File: www/install/steps/process/configFileSetup.php [hunk 5]
```diff
@@ -106,16 +107,20 @@
     '/--HTTPSERVERADDRESS--/',
     '/--HTTPSERVERPORT--/',
     '/--SSLMODE--/',
-    '/--CENTREON_TRAPDIR--/'
+    '/--CENTREON_TRAPDIR--/',
+    '/--ENGINE_COMMAND--/',
+    '/--GORGONE_VARLIB--/'
```
Comment (sc979) [inline www/install/steps/process/configFileSetup.php:112]: 111 and 112 revert

File: www/install/php/Update-20.04.1.php [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+    if (!file_exists($destinationFile) || 0 === filesize($destinationFile)) {
+        $errorMessage = 'Gorgone configuration file is not created properly';
+        throw new \InvalidArgumentException($errorMessage);
+    }
+} catch (\Exception $e) {
+    $centreonLog->insertLog(
+        4,
+        $versionOfTheUpgrade . $errorMessage .
+        " - Code : " . (int)$e->getCode() .
```
Comment (kduret) [inline www/install/php/Update-20.04.1.php:151]:
```suggestion
        " - Code : " . $e->getCode() .
```

File: www/install/php/Update-20.04.1.php [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+        " - Error : " . $e->getMessage() .
+        " - Trace : " . $e->getTraceAsString()
+    );
+    throw new \Exception($versionOfTheUpgrade . $errorMessage, (int)$e->getCode(), $e);
```
Comment (kduret) [inline www/install/php/Update-20.04.1.php:155]:
```suggestion
    throw new \Exception($versionOfTheUpgrade . $errorMessage, $e->getCode(), $e);
```

File: www/install/php/Update-20.04.1.php [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+/**
+ * Queries needing exception management BUT no rollback if failing
+ */
+try {
+    /*
+     * Get user data to generate a new config file for the gorgone daemon module
+     */
+
+    // get current configuration data
+    require_once __DIR__ . '/../../../config/centreon.config.php';
```
Comment (kduret) [inline www/install/php/Update-20.04.1.php:38]: require should be at the top of file

File: www/install/php/Update-20.04.1.php [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+    if (!file_exists($fileTpl) || 0 === filesize($fileTpl)) {
+        $errorMessage = 'Gorgone configuration template is empty or missing';
+        throw new \InvalidArgumentException($errorMessage);
+    }
+    $content = file_get_contents($fileTpl);
+    $content = preg_replace($pattern, $userValues, $content);
+    $destinationFolder = _CENTREON_ETC_ . '/../centreon-gorgone';
+    $destinationFile = returnFinalFileName($destinationFolder . '/config.d/40-gorgoned.yaml');
+
+    // checking if mandatory /etc/centreon-gorgone and sub-folder already exists
```
Comment (sc979) [inline www/install/php/Update-20.04.1.php:135]:
```suggestion
    // checking if mandatory centreon-gorgone configuration sub-folder already exists
```

**Pull request merged**